fix(bedrock_agent): fix querystring field resolution

matteofigus opened the pull request:
**Issue number:** #6520

## Summary

### Changes

This PR fixes an issue where Bedrock Agent parameters containing commas (like SQL queries) were being truncated because the  method was splitting them by commas.

The fix overrides the  method in the  class to preserve parameter values without splitting them by commas. This ensures that parameters like SQL queries with commas are correctly passed to the handler function.

Added unit and functional tests to verify the fix works correctly.

### User experience

**Before:**
When using a Bedrock Agent with parameters containing commas (like SQL queries), only the part before the first comma was being passed to the handler function.

For example, with a parameter value of "SELECT a.source_name, b.thing FROM table", only "SELECT a.source_name" would be passed to the handler.

**After:**
The entire parameter value is correctly passed to the handler function, including any commas.

For example, with a parameter value of "SELECT a.source_name, b.thing FROM table", the entire string is passed to the handler.

## Checklist

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

matteofigus committed:
fix: prevent splitting Bedrock Agent parameters with commas

This fixes issue #6520 where Bedrock Agent parameters containing commas (like SQL queries) were being truncated because the resolved_query_string_parameters method was splitting them by commas.

The fix overrides the resolved_query_string_parameters method in the BedrockAgentEvent class to preserve parameter values without splitting them by commas.

matteofigus committed:
style: format test files

leandrodamascena commented on aws_lambda_powertools/utilities/data_classes/bedrock_agent_event.py:
```diff
@@ -112,6 +112,19 @@ def query_string_parameters(self) -> dict[str, str]:
         parameters = self.get("parameters") or []
         return {x["name"]: x["value"] for x in parameters}
 
+    @cached_property
```
I know we are using `cached_property` in the common base class, but tbh it doesn't have any effect here. We just access the headers once per request and when a new request arrives we create a new event source data class object, so, I'd prefer to make it `@property` to not save anything in cache.

Pls remove in the base class too https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/utilities/data_classes/common.py#L172

leandrodamascena commented on tests/functional/event_handler/_pydantic/test_bedrock_agent.py:
```diff
@@ -343,3 +343,48 @@ def handler() -> Optional[Dict]:
+    def run_sql_query(query: Annotated[str, Query()]):
+        nonlocal received_query
+        received_query = query
+        return {"result": "Query executed"}
```
Just return `query` parameter and assert it. You don't need to have global/local variables here, it make the test unclear.

leandrodamascena commented on tests/functional/event_handler/_pydantic/test_bedrock_agent.py:
```diff
@@ -343,3 +343,48 @@ def handler() -> Optional[Dict]:
+    result = app(event, {})
+
+    # THEN the parameter with commas should be correctly passed to the handler
+    assert received_query == "SELECT a.source_name, b.thing FROM table"
```
After changing the return of the function, assert `body["result"]` here.

leandrodamascena commented on tests/unit/data_classes/required_dependencies/test_bedrock_agent_event.py:
```diff
@@ -62,3 +62,35 @@ def test_bedrock_agent_event_with_post():
     assert properties[1].value == raw_properties[1]["value"]
+
+
+def test_bedrock_agent_event_with_comma_parameters():
```
You can remove this test.

leandrodamascena reviewed (changes requested):
Hi @matteofigus, thanks for submitting this PR! This solves the problem for BedrockResolver, but in other resolvers where we have multiquerystring fields, we have the same problem. For example, if you replace Bedrock Resolver with `APIGatewayHttpResolver` (of course, changing the payload), you will observe the same problem.

So, with that said, do you want to open a new issue to investigate other resolvers or do you want to address the issue in this same PR? I would prefer the second option, but feel free to make your own decision.

matteofigus committed:
fix: address PR feedback for Bedrock Agent parameters with commas

- Change @cached_property to @property for resolved_query_string_parameters
- Simplify the functional test by directly returning the query parameter
- Remove redundant unit test

leandrodamascena committed:
Merge branch 'develop' into fix-bedrock-agent-comma-parameters

leandrodamascena reviewed (approved):
Thank you @matteofigus! APPROVED
